install/playbooks/main-inline.yml

```diff
@@ -7,72 +7,113 @@
     - '{{ playbook_dir }}/../../config/defaults.yml'
   roles:
     - role: load-defaults
+      tags: always
     - role: external-ip-type
+      tags: always
     - role: packages
+      tags: apt
     - role: system-prepare
+      tags: system
     - role: remote-access
+      tags: ssh
     - role: luks-remote
+      tags: luks
     - role: nginx
+      tags: nginx
     - role: dns-server-bind
       when: bind.install
+      tags: bind
     - role: dns-server-check-propagation
       when: bind.install and bind.propagation.check
+      tags: bind
     - role: certificates-all
+      tags: ssl
```

Comment:
`certificates` for the tag?
Or `tls` if you want, but I see it more as about the certificates than the TLS configuration.
In any case, "SSL" is long gone now. :sweat_smile:

As it is, the `certificates-all` role should either be:
- tagged `always` and the different role tags be handled in the role (applied to the import_tasks create_one) if possible ;
- tagged with every tag needing a certificate, plus the `certificates` one.

Another way would be to keep several invocations with parameters, as it is today, each tagged with `certificates` and the tag of the role needing the certificate. It could be either in the play, or kept in role dependencies (sort of "hidden").

Just from looking at it, I think:
- the solution with the `always` tag should be avoided (for not handling the tags inside the role) ;
- I might prefer the certificate declaration to be visible (even if it might be a good fit for dependencies) ;
- the all-tags-on-certificates-all seems fine ;
- I might prefer the split version so that it is clear where and how to add a certificate for a new role added to the play (but then it might be a bit less readable with those vars declared in the play).

Well, one of the last two would be great, I think.

Reply:
Ok, outdated because the move to main.yml wasn't picked up. Applies to main.yml, now.

The tag on extra-certs should be `certficates` too. It is used already in the `extra-certs.yml` playbook.

Reply:
Yes, this is a huge work in progress. but I am happy with your suggestions, and even more that you are following ! I will continue in the afternoon

Reply:
in our case, ssl was more a tag chosen because it is short than anything else. As a short and quick tag, tls is fine too. Now, I had a look, and it seems we are using "cert" and "certificates", I will come back to this tomorrow or later.
The reason why I moved them all in one place is performance. This was the second role that was taking most of the time.

Reply:
`certificates` or `cert` is fine by me. I usually choose the longer forms, but that's just me.
The role is doing a lot of work, indeed.